tests/integration/targets/aci_fabric_setup_pod/tasks/main.yml

```diff
@@ -0,0 +1,168 @@
+    timeout: 10
+  delegate_to: localhost
+  when: cleaned is changed
+
+# CREATE FABRIC SETUP POD
+- name: Create ansible_fabric_pod_setup
+  aci_fabric_setup_pod:
+    <<: *aci_info
+    podId: 77
+    pool: 10.77.0.0/16
+    description: Ansible test pod setup
+    state: present
+  register: add_fabric_pod_setup
+
+- name: Sleep for 10 seconds and continue
+  ansible.builtin.wait_for:
+    timeout: 10
+  delegate_to: localhost
+
+- name: Verify profile creation
+  assert:
+    that:
+    - add_fabric_pod_setup is changed
+    - add_fabric_pod_setup.current.0.fabricSetupP.attributes.dn == "uni/controller/setuppol/setupp-77"
+    - add_fabric_pod_setup.current.0.fabricSetupP.attributes.podId == "77"
+    - add_fabric_pod_setup.current.0.fabricSetupP.attributes.podType == "physical"
+    - add_fabric_pod_setup.current.0.fabricSetupP.attributes.tepPool == "10.77.0.0/16"
+    - add_fabric_pod_setup.current.0.fabricSetupP.attributes.descr == "Ansible test pod setup"
+    - add_fabric_pod_setup.current.0.fabricSetupP.attributes.annotation == 'orchestrator:ansible'
+
+# CREATE FABRIC SETUP POD AGAIN TO TEST IDEMPOTENCE
+- name: Create ansible_fabric_pod_setup again
+  aci_fabric_setup_pod:
+    <<: *aci_info
+    podId: 77
+    pool: 10.77.0.0/16
+    description: Ansible test pod setup
+    state: present
+  register: add_fabric_pod_setup_again
+
+- name: Verify profile creation idempotence
+  assert:
+    that:
+    - add_fabric_pod_setup_again is not changed
+    - add_fabric_pod_setup_again.current.0.fabricSetupP.attributes.dn == "uni/controller/setuppol/setupp-77"
+    - add_fabric_pod_setup_again.current.0.fabricSetupP.attributes.podId == "77"
+    - add_fabric_pod_setup_again.current.0.fabricSetupP.attributes.podType == "physical"
+    - add_fabric_pod_setup_again.current.0.fabricSetupP.attributes.tepPool == "10.77.0.0/16"
+    - add_fabric_pod_setup_again.current.0.fabricSetupP.attributes.descr == "Ansible test pod setup"
+
+# QUERY FABRIC SETUP POD
+- name: query fabric setup pod
+  aci_fabric_setup_pod:
+    <<: *aci_info
+    podId: 77
+    state: query
+  register: query_pod_profile
+
+- name: Verify profile query
+  assert:
+    that:
+    - query_pod_profile is not changed
+    - query_pod_profile.current.0.fabricSetupP.attributes.dn == "uni/controller/setuppol/setupp-77"
+    - query_pod_profile.current.0.fabricSetupP.attributes.podId == "77"
+    - query_pod_profile.current.0.fabricSetupP.attributes.podType == "physical"
+    - query_pod_profile.current.0.fabricSetupP.attributes.tepPool == "10.77.0.0/16"
+    - query_pod_profile.current.0.fabricSetupP.attributes.descr == "Ansible test pod setup"
+
+# QUERY ALL POD SETUPS
+- name: query all fabric pod setups
+  aci_fabric_setup_pod:
+    <<: *aci_info
+    state: query
+  register: query_pod_profile_all
+
+- name: Verify profile query idempotence
+  assert:
+    that:
+    - query_pod_profile_all is not changed
+
+# DELETE POD SETUP
+- name: Delete pod setup
```

Comment:
Please add test with checkmode and assert proposed and previous for the test.

Reply:
Done.